dbms/src/Storages/KVStore/MultiRaft/RaftCommands.cpp

```diff
@@ -348,6 +348,8 @@ std::pair<EngineStoreApplyRes, DM::WriteResult> Region::handleWriteRaftCmd(
 
     auto is_v2 = this->getClusterRaftstoreVer() == RaftstoreVer::V2;
 
+    std::vector<TiKVKey> deleting_lock_keys;
+    deleting_lock_keys.reserve(cmds.len / 2); // write + lock
```

Comment:
Could also be:
- lots of locks put, e.g. big txn
- some default cfs, e.g. large value